xbmc/guilib/TextureFormats.h

```diff
@@ -25,4 +25,158 @@ enum XB_FMT
   XB_FMT_MASK        = 0xFFFF,
   XB_FMT_OPAQUE      = 0x10000,
 };
+
+enum KD_TEX_FMT
+{
+  KD_TEX_FMT_UNKNOWN            = 0x0000,
+
+  // Legacy XB_FMT formats family
+  KD_TEX_FMT_LEGACY             = 0x0000,
+
+  // SDR texture family
+  KD_TEX_FMT_SDR                = 0x1000,
+  KD_TEX_FMT_SDR_R8             = 0x1000, // 8bpp, single channel
+  KD_TEX_FMT_SDR_RG8            = 0x1100, // 16bpp, dual channel
+  KD_TEX_FMT_SDR_R5G6B5         = 0x1200, // 16bpp, 5/6 bit per color channel
+  KD_TEX_FMT_SDR_RGB5_A1        = 0x1300, // 16bpp, 5 bit per color channel, pt-alpha
+  KD_TEX_FMT_SDR_RGBA4          = 0x1400, // 16bpp, 4 bit per channel
+  KD_TEX_FMT_SDR_RGB8           = 0x1500, // 24bpp, 8 bit per channel, no alpha (unsuitable for GPUs!)
```

Comment:
What is the use of specifying the hex values for all the enums? They seem to have been chosen to not overlap, why?

Reply:
They are used in 32 bit fields of our XBT file to describe texture type and opaqueness.

Reply:
I'm adding new flags for color space, gamma and similar. 

The textures are grouped in families, which allow for an easy query of supported formats. Different GPUs support different schemes. For example:
- Utgard/VC4: SDR, ETC1
- early Midgard: SDR, HDR, ETC1, ETC2
- later Midgard/Bifrost SDR, HDR, ETC1, ETC2, ASTC
- VC6: SDR, HDR, ETC1, ETC2, ASTC, S3TC, RGTC

Also, some devices might not have native sRGB sampler/framebuffer (VC4).

Reply:
I must be missing something. I understand defining every possible useful format for any gpu internally in Kodi and adding primaries/transfer/etc metadata to the textures,

I don't understand how allowing that many formats in the XBT could be a good thing, it will require a huge number of conversion functions and couple Kodi's internals.

Does XBT allow any format or only a small subset (which would make more sense to me)? Same question about the new col space/gamma, only certain combinations make sense, and only combined with certain formats. In which case, internal and external format definitions should be separate, with mapping functions.

For the XBT file format, I don't know the details but understand there could be a need to pack more bits into existing fields for compatibility reasons, but Kodi's internal structures don't have that constraint. The gamma uses the last available nibble for a 32bit int, what happens next?

Reply:
> I don't understand how allowing that many formats in the XBT could be a good thing, it will require a huge number of conversion functions and couple Kodi's internals.

There is no need for conversion functions. On the contrary, such would defeat the core idea of having the XBT format.

The way I picture it, texture compression formats should always be accompanied by raw, uncompressed equivalents, in case the GPU can't use the compressed version. In the future, we could distribute tailored texture packs, reflecting the device capabilities. 

> Does XBT allow any format or only a small subset (which would make more sense to me)?

XBT is everything we want it to be. It doesn't make sense to me to limit it to a subset of what Kodi might support.

> Same question about the new col space/gamma, only certain combinations make sense, and only combined with certain formats. In which case, internal and external format definitions should be separate, with mapping functions.

Yes, some permutations of the format don't make sense or a problematic. For example, there is no explicit sRGB version of an ETC1 compressed texture (it would decode to the GPU gamma). Also, using a HLG gamma curve with a BT709 color space makes little sense. Not every combination should be used or is without caveats.

But most combinations make sense. And there are some unusual ones which make sense. For example, you could have a compressed ASTC texture in BT2020 color space, using the sRGB gamma curve. 

And the mapping is already there. Unsupported formats are silently dropped. The colorspace is expected to be BT709, with a sRGB gamma 

Also, don't focus only on our XBT file format. Right now, image decoders can only pass RGBA8 textures (at most) to the core. The decoder has to convert/scale the image, which is far from ideal. We need a way to convey the image properties, even if they can't be processed by the GPU, so that we can convert them to a suitable format/size. 

> For the XBT file format, I don't know the details but understand there could be a need to pack more bits into existing fields for compatibility reasons, but Kodi's internal structures don't have that constraint.

Right now, the image decoder interface has the very same constraint. And up to this PR, there has been the constraint in the very files this PR touches. The XBT_FMT is everywhere.

This PR just serves as a starting point for me. Otherwise, I'd rip out most of the texture processing, probably driving me insane while trying to get to the current feature set. I'm all ears if you have any suggestions how to proceed, but this seemed the most logical place to me.

> The gamma uses the last available nibble for a 32bit int, what happens next?

We can always introduce a new XBT version breaking backwards compatibility. But I'd like to avoid that at the current stage, as it is not needed. Let's cross that bridge when we have to.